app/models/time_period.rb

```diff
@@ -27,4 +27,21 @@ def start_month
   def end_month
     YearMonth.new(ends_on.year, ends_on.month)
   end
+
+  # Obtain the duration between the start and end of the time period
+  # in months. This is the number of months covered by the range.
+  def duration
+    (@ends_on.year * 12 + @ends_on.month) - (@starts_on.year * 12 + @starts_on.month) + 1
+  end
+
+  # Retrieve the time period that precedes the current period.
+  # This time period should match the month range so if current period is
+  # Jan-Mar 2017 the previous period is Jan-Mar 2016. If current period is
+  # Oct 2016 - Feb 2017 then the previous period is Oct 2015 - Feb 2016.
+  def previous_period
+    TimePeriod.new(
+      @starts_on.dup.advance(months: -12),
```

Comment:
this works, but it doesn't quite read right to me to have an advance method that goes back in time. Could use `@ends_on.dup.months_since(-12)` instead?

Reply:
Discussed this on slack, and neither are really that readable, would prefer a method that was the opposite of advance instead....